app.rb

```diff
@@ -0,0 +1,65 @@
+require './lib/models'
+
+require 'bcrypt'
+require 'pg'
+require 'sinatra/base'
+require 'sinatra/flash'
+
+class Chitter < Sinatra::Base
+  enable :sessions
+  register Sinatra::Flash
+
+  get '/' do
+    @user = User.get(session[:id])
+    @peeps = Peep.all(:order => [:id.desc])
+    erb(:index)
+  end
+
+  get '/users/new' do
+    erb(:'users/new')
+  end
+
+  post '/users' do
+    user = User.create(
+      peeper: params['name'],
+      username: params['username'],
+      email: params['email'],
+      password: BCrypt::Password.create(params['password'])
+    )
+
+    if user.errors.size.positive?
```

Comment:
does `if user.errors` work?

Reply:
another pattern, which you'll see used quite often, is to separate instantiation from entering a new record in the DB (`create` combines both).  Something like this (example taken from a Rails controller):
```ruby
def create
    @client = Client.new(params[:client])
    if @client.save
      redirect_to @client
    else
      # This line overrides the default rendering behavior, which
      # would have been to render the "create" view.
      render "new"
    end
  end
```

Reply:
Hi @EdwardAndress ! Thank you for your thorough code review, finally I have some time to go over things more focused. 

In order:
1) Yes, exactly, `user.errors` works fine.

2) Do you think we can discuss your second comment on this section at some point next week? I understand the difference between **Create** and **New** (which requires save to store it into the DB apart from creating a new instance of an object).

I would like to address especially the advantage of separating these cases in this specific example. Thanks in advance!